pinot-broker/src/main/java/org/apache/pinot/broker/broker/AllowAllAccessControlFactory.java

```diff
@@ -34,6 +36,10 @@ public AllowAllAccessControlFactory() {
   public void init(PinotConfiguration configuration) {
   }
 
+  public void init(ZkHelixPropertyStore<ZNRecord> propertyStore) {
+  }
+
+  @Override
```

Comment:
see above. if there's any change in this file, it's not backwards-compatible

Reply:
@apucher Thanks for your suggestions. I have changed the signature of the static loadFactory to include both configuration and property store. 

Also I have reconstructed AccessControlFactory and relevant class about Controller/Broker/Server. And introduce an init(PinotConfiguration, ZkHelixPropertyStore) that defaults to invoking the legacy init(PinotConfiguration) method Rather that using instanceof. Looking forward your review again.

Reply:
you're still breaking backwards-compatibility. if you have to make changes to this file, you equally affect 3rd party implementations which aren't in the open source repository.

you can simply move the implementation of `init(PinotConfiguration, ZkHelixPropertyStore)` to the AccessControlFactory parent class and declare it non-abstract. then, override the method in you new implementation while this class (and similar ones) can remain unchanged

Reply:
Thanks for your detailed explanation. I have accepted your above advice. And undo any changes in this file to keep consistency with the original file.  
Please check again.